src/Collection.php

```diff
@@ -539,10 +529,7 @@ public static function fromResource($resource): self
      */
     public static function fromString(string $string, string $delimiter = ''): self
     {
-        return new self(
-            static fn (string $string, string $delimiter): Iterator => new StringIterator($string, $delimiter),
-            [$string, $delimiter]
-        );
+        return new self(static fn (): Iterator => new StringIterator($string, $delimiter));
```

Comment:
does it matter if we pass the parameter to the closure or we just let it capture it from the environment? At least with the testing we have it doesn't seem to make any difference, but it fixes a PHPStan error

Reply:
Sure technically this is valid, it's just confusing according to me, this is why I made the change.

I wish I could understand why PHPStan doesn't like it.